core/prelude/types/uint.carbon

```diff
@@ -30,4 +30,147 @@ impl forall [N:! IntLiteral()] UInt(N) as As(IntLiteral()) {
   fn Convert[self: Self]() -> IntLiteral() = "int.convert_checked";
 }
 
-// TODO: Operations.
+impl forall [N:! IntLiteral()] UInt(N) as Add {
+  fn Op[self: Self](other: Self) -> Self = "int.sadd";
+}
+// TODO: Once we can lower specific functions, change this to:
+// impl forall [N:! IntLiteral()] UInt(N) as AddAssign {
+impl u32 as AddAssign {
+  fn Op[addr self: Self*](other: Self) {
+    *self = *self + other;
+  }
+}
+// TODO: Once we can lower specific functions, change this to:
+// impl forall [N:! IntLiteral()] UInt(N) as Inc {
+impl u32 as Inc {
```

Comment:
FWIW, I think some of these spacing choices don't make as much sense to me... Like I get the choice to avoid whitespace between `BitAnd` and `BitAndAssign`, but why isn't `Inc` just on its own, and similar for `Dec`? I don't really view these as associated with `Add`/`Sub` -- sort of how `Sub` could be written in terms of `Add` and `Negate` (or `Negate` in terms of `Sub`), but are treated as distinct.

Reply:
This is pre-existing, but IIRC the idea here was that they are grouped together because `Inc` is implemented in terms of `AddAssign`, so must be ordered after `AddAssign` -- the groups with blank lines between them can be put in any order (and are ordered alphabetically by the first interface name in the group) but should be kept as a group if they are reordered due to dependencies.

Perhaps this was more obvious when the `Inc` impl actually named the preceding `AddAssign` impl rather than implicitly using it via an operator.

Reply:
As an alternative, we could order these as:
- basic operators, alphabetically
- `*Assign` operators, alphabetically
- `Inc`/`Dec` operators, alphabetically
Maybe that'd be nicer?